test/test_mp_input_sharding.py

```diff
@@ -0,0 +1,98 @@
+import sys
+import numpy as np
+import unittest
+
+import torch
+import torch_xla
+from torch_xla import runtime as xr
+import torch_xla.core.xla_model as xm
+from torch_xla.distributed.spmd import Mesh
+import torch_xla.distributed.spmd as xs
+import torch_xla.distributed.parallel_loader as pl
+
+xr.use_spmd()
+
+
+class MpInputShardingTest(unittest.TestCase):
+
+  class fake_dataloader:
+
+    def __init__(self, batch):
+      self.batch = batch
+
+    def __iter__(self):
+      return self
+
+    def __next__(self):
+      return self.batch
+
+  @unittest.skipUnless(xr.global_runtime_device_count() > 1,
+                       "Multiple devices required for tupled partition spec")
+  def test_multiple_inputs(self):
+    device = xm.xla_device()
+    batch = {'x': torch.randn((16, 128)), 'y': torch.randn((16, 128, 128))}
+    train_loader = self.fake_dataloader(batch)
+    num_devices = xr.global_runtime_device_count()
+    mesh_shape = (num_devices, 1)
+    device_ids = np.arange(num_devices)
+    mesh = Mesh(device_ids, mesh_shape, ('x', 'y'))
```

Comment:
fyi you can use https://github.com/pytorch/xla/blob/master/torch_xla/distributed/spmd/xla_sharding.py#L163-L189 directly